server/dao.js

```diff
@@ -73,3 +71,55 @@ exports.getProductsByCategory = (category) => {
     });
   });
 };
+
+exports.insertOrder = async (order, id_array, quantity_array) => {
+  //Need to iterate over different products in list
+  var i = 0;
+  const promiseList = [];
+  for (const id of id_array) {
+    promiseList.push(createInsertOrderPromise(order, id, quantity_array[i++]));
+  }
+  return Promise.all(promiseList)
+    .then()
+    .catch((err) => console.log(err));
+};
+
+function createInsertOrderPromise(order, id, quantity) {
+  return new Promise((resolve, reject) => {
+    const sql =
+      "INSERT INTO ORDERS (order_id, ref_product, ref_user, date_order, quantity, status) VALUES (?, ?, ?,?,?,?)";
+    db.run(
+      sql,
+      [
+        order.order_id,
+        id,
+        order.ref_user,
+        order.date_order,
+        quantity,
+        "pending",
+      ],
+      function (err) {
+        if (err) {
+          resolve(false);
+        } else {
+          console.log("Succesfully added product " + id + " in order n°" + order.order_id);
+          resolve(true);
+        }
+      }
+    );
+  })
+}
+
+exports.deleteTestOrder = function () {
+  return new Promise((resolve, reject) => {
+    const sql = "DELETE from ORDERS where order_id = 0";
```

Comment:
`where order_id = ?`
Receiving the 0 from outside, so that this method can be used also later.